demo-terraform/aws/serverless-app/modules/create-api-gateway-resource/main.tf

```diff
@@ -1,64 +1,64 @@
 
 resource "aws_api_gateway_resource" "resource" {
-    rest_api_id = var.rest_api_id
-    parent_id   = var.parent_id
-    path_part   = var.path_part
+  rest_api_id = var.rest_api_id
+  parent_id   = var.parent_id
+  path_part   = var.path_part
 }
 
 ## OPTION METHOD to enable CORS support
 resource "aws_api_gateway_method" "resource_options" {
-    count = var.enable_cors ? 1 : 0
+  count = var.enable_cors ? 1 : 0
 
-    rest_api_id = var.rest_api_id
-    resource_id = aws_api_gateway_resource.resource.id
-    http_method   = "OPTIONS"
-    authorization = "NONE"
+  rest_api_id   = var.rest_api_id
+  resource_id   = aws_api_gateway_resource.resource.id
+  http_method   = "OPTIONS"
+  authorization = "<authorization>"
```

Comment:
**Accurics remediation**
 
**[HIGH]: Ensure authorization is enabled for AWS API Gateway Method**
 
**Vulnerability**
AWS API Gateway Method has authorization disabled which may lead to unauthorized access.
 
**Remediation**
AWS API Gateway Method should have authorization enabled. Authorization can be done by using multiple methods such as AWS IAM or AWS Cognito User Pools or by using custom authorizer using AWS Lambda.

source: accurics-